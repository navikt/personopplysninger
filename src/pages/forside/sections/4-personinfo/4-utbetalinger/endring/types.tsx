import { OptionType } from '../../../../../../types/option';

export interface OutboundNorskKontonummer {
    value: string;
}

export interface OutboundUtenlandsbankonto {
    value: string | null;
    utenlandskKontoInformasjon: {
        landkode: string;
        landkodeTobokstavs?: string;
        valuta: string;
        swift?: string;
        bank: {
            adresseLinje1?: string;
            adresseLinje2?: string;
            adresseLinje3?: string;
            kode?: string;
            navn: string;
        };
    };
}

export interface FormFields {
    kontonummer: string;
    land?: OptionType;
    valuta?: OptionType;
    banknavn: string;
    kontonummerIban: string;
    bickode: string;
    retningsnummer: string;
    bankkode: string;
    adresse1: string;
    adresse2: string;
    adresse3: string;
}
