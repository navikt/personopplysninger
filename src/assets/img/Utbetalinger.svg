<?xml version="1.0" encoding="UTF-8"?>
<svg width="34px" height="34px" viewBox="0 0 34 34" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 55 (78076) - https://sketchapp.com -->
    <title>Color/32-content/box-2 Copy@1x</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <rect id="path-1" x="0" y="0" width="33.6" height="33.6"></rect>
    </defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="GDPR-Personopplysning" transform="translate(-309.000000, -4868.000000)">
            <g id="Color/24-business-finance/coin-receive" transform="translate(309.000000, 4868.000000)">
                <mask id="mask-2" fill="white">
                    <use xlink:href="#path-1"></use>
                </mask>
                <g id="Background"></g>
                <path d="M6.29999993,18.2 L0.69999993,18.2 C0.31359993,18.2 -7.00000555e-08,18.5320471 -7.00000555e-08,18.9411765 L-7.00000555e-08,30.0588235 C-7.00000555e-08,30.4679529 0.31359993,30.8 0.69999993,30.8 L6.29999993,30.8 C6.68499993,30.8 6.99999993,30.4679529 6.99999993,30.0588235 L6.99999993,18.9411765 C6.99999993,18.5320471 6.68499993,18.2 6.29999993,18.2 Z" id="Fill/coin-receive" fill="#669DB4" fill-rule="evenodd" mask="url(#mask-2)"></path>
                <g id="Group" stroke-width="1" fill-rule="evenodd" mask="url(#mask-2)">
                    <g transform="translate(18.200000, 0.000000)" fill-rule="nonzero" id="Path">
                        <path d="M4.9,9.8 C7.602,9.8 9.8,7.602 9.8,4.9 C9.8,2.1966 7.602,0 4.9,0 C2.198,0 0,2.1966 0,4.9 C0,7.602 2.198,9.8 4.9,9.8 Z" fill="#FFBD66"></path>
                        <path d="M4.2,3.5 C4.2,3.1136 4.5136,2.8 4.9,2.8 C5.285,2.8 5.6,3.1136 5.6,3.5 L5.6,6.3 C5.6,6.6864 5.285,7 4.9,7 C4.5136,7 4.2,6.6864 4.2,6.3 L4.2,3.5 Z" fill="#B26D14"></path>
                    </g>
                </g>
                <g id="Group-Copy" stroke-width="1" fill-rule="evenodd" mask="url(#mask-2)">
                    <g transform="translate(9.800000, 8.400000)" fill-rule="nonzero" id="Path">
                        <path d="M4.9,9.8 C7.602,9.8 9.8,7.602 9.8,4.9 C9.8,2.1966 7.602,0 4.9,0 C2.198,0 0,2.1966 0,4.9 C0,7.602 2.198,9.8 4.9,9.8 Z" fill="#FFBD66"></path>
                        <path d="M4.2,3.5 C4.2,3.1136 4.5136,2.8 4.9,2.8 C5.285,2.8 5.6,3.1136 5.6,3.5 L5.6,6.3 C5.6,6.6864 5.285,7 4.9,7 C4.5136,7 4.2,6.6864 4.2,6.3 L4.2,3.5 Z" fill="#B26D14"></path>
                    </g>
                </g>
                <path d="M14.8970199,25.5532562 C14.7916862,25.7202866 14.8417012,25.9410813 15.0087316,26.046415 C15.0243355,26.0562553 15.0406775,26.0648731 15.0576114,26.0721914 L19.6,28.0352838 L28,25.0160726 C29.9050585,24.4738536 31.7010598,23.7256058 33.3890134,25.4037664 C33.5514826,25.5668613 33.6262471,25.7986277 33.5917404,26.0261022 C33.5543581,26.2550073 33.4105801,26.4495766 33.2035398,26.5525839 C29.2309538,28.5283212 26.5609964,30.0462482 24.6113668,31.1550073 C21.6653556,32.8303066 20.2548935,33.6 18.6057599,33.6 C17.2226155,33.6 15.671251,33.057781 12.8992112,32.0548905 C11.4901868,31.544146 9.73609527,30.910365 7.49172075,30.1649927 C7.19841363,30.067708 7,29.7944526 7,29.4868613 L7,20.3098592 C7,19.9180169 7.35689563,19.6 7.79664202,19.6 L10.1865681,19.6 C12.3528749,19.6 14.0162489,20.2969485 15.1637185,21 L21.7,21 C22.859798,21 23.8,21.940202 23.8,23.1 C23.8,24.259798 22.859798,25.2 21.7,25.2 L15.5374257,25.2 C15.2774671,25.2 15.0356862,25.3333694 14.8970199,25.5532562 Z" id="Path" fill="#FFE5C2" fill-rule="evenodd" mask="url(#mask-2)"></path>
            </g>
        </g>
    </g>
</svg>