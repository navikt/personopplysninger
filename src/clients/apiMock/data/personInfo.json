{
  "personalia": {
    "fornavn": "DONALD FAUNTLEROY",
    "etternavn": "DUCK",
    "personident": {
      "verdi": "12345678911",
      "type": "FNR"
    },
    "utenlandskbank": {
      "adresse1": "Gävlegatan 22",
      "adresse2": "113 30 Stockholm",
      "adresse3": "",
      "bankkode": "90081210188",
      "banknavn": "Nordax Bank AB",
      "iban": "",
      "kontonummer": "US90081210188",
      "land": "Sverige",
      "swiftkode": "HELGSE21",
      "valuta": "Tugriks Mongolia"
    },
    "tlfnr": {
      "jobb": "12121212",
      "mobil": "24242424",
      "privat": "36363636",
      "datakilder": [{}, {}]
    },
    "spraak": "NB",
    "epostadr": "TODO",
    "personstatus": "BOSA",
    "statsborgerskap": "NOR",
    "foedested": "NOR",
    "sivilstand": "UGIF",
    "kjoenn": "M",
    "datakilder": [{}, {}, {}, {}]
  },
  "enhetKontaktInformasjon": {
    "enhet": {
      "postadresse": {
        "type": "stedsadresse",
        "postnummer": "0614",
        "poststed": "Oslo",
        "gatenavn": "Slottsveien",
        "husnummer": "22",
        "husbokstav": "A"
      },
      "publikumsmottak": [
        {
          "gateadresse": "Peder Falcks veg",
          "poststed": "Trondheim",
          "husnummer": "31",
          "husbokstav": null,
          "postnummer": "7044",
          "stedsbeskrivelse": null,
          "aapningMandag": {
            "dag": "Mandag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": "Timeavtaler også utenom åpningstiden"
          },
          "aapningTirsdag": {
            "dag": "Tirsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningOnsdag": {
            "dag": "Onsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningTorsdag": {
            "dag": "Torsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningFredag": {
            "dag": "Fredag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningAndre": [
            {
              "dag": "15.05.2019",
              "fra": "09:00",
              "til": "12:00",
              "stengt": "false",
              "kommentar": "Her kan det også være en merknad "
            },
            {
              "dag": "17.05.2019",
              "fra": null,
              "til": null,
              "stengt": "true"
            },
            {
              "dag": "29.05.2019",
              "fra": "09:00",
              "til": "13:30",
              "stengt": "false",
              "kommentar": "Onsdag før Kristi himmelfartsdag"
            }
          ]
        },
        {
          "gateadresse": "Kirkegata",
          "poststed": "Trondheim",
          "husnummer": "25",
          "husbokstav": null,
          "postnummer": "7014",
          "stedsbeskrivelse": null,
          "aapningMandag": {
            "dag": "Mandag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": "Timeavtaler også utenom åpningstiden"
          },
          "aapningTirsdag": {
            "dag": "Tirsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningOnsdag": {
            "dag": "Onsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningTorsdag": {
            "dag": "Torsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningFredag": {
            "dag": "Fredag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningAndre": [
            {
              "dag": "15.05.2019",
              "fra": "09:00",
              "til": "12:00",
              "stengt": "false",
              "kommentar": "Her kan det også være en merknad "
            },
            {
              "dag": "17.05.2019",
              "fra": null,
              "til": null,
              "stengt": "true"
            },
            {
              "dag": "29.05.2019",
              "fra": "09:00",
              "til": "13:30",
              "stengt": "false",
              "kommentar": "Onsdag før Kristi himmelfartsdag"
            }
          ]
        }
      ],
      "tlfperson": "55553333",
      "spesielleopplysninger": "Timeavtale 8-15, man - fre."
    }
  },
  "adresser": {
    "boadresse": {
      "adressetillegg": "C/O Rune Aremark",
      "adresse": "Bjørvika 4D",
      "datoFraOgMed": "2017-02-21",
      "kommune": "1239",
      "land": "Norge",
      "matrikkeladresse": {},
      "postnummer": "1239",
      "poststed": "Jepp",
      "veiadresse": {
        "bokstav": "A",
        "bolignummer": "H0102",
        "gatekode": "01239",
        "husnummer": "11"
      }
    },
    "geografiskTilknytning": {
      "bydel": "500102",
      "datoFraOgMed": "2012-10-01",
      "kommune": null,
      "land": null,
      "enhet": "NAV Falkenborg"
    },
    "postadresse": {
      "adresse1": "12345 HFJEF IQUFJ HFURHGHQBEUQ",
      "adresse2": "BATOOM LUNCHAGRAMA,GONGRUPEA,DD",
      "adresse3": "GGLAM,10000 HCAMGHU,THAILAND",
      "datoFraOgMed": "2018-03-13",
      "land": "THA",
      "postnummer": "",
      "poststed": "Nittedal"
    },
    "prioritertAdresse": "TIAD",
    "tilleggsadresse": {
      "adresse1": "JEPP 10jfioew JIO LPFL ",
      "adresse2": "LOOOL FKEOWIF JEWIOF WECOOOO FJEI",
      "adresse3": "",
      "bolignummer": "",
      "bydel": "",
      "datoFraOgMed": "2019-01-12",
      "datoTilOgMed": "2023-01-11",
      "gateKode": "0270",
      "husbokstav": "",
      "husnummer": "0020",
      "kommunenummer": "0131",
      "postboksanlegg": "",
      "postboksnummer": "",
      "postnummer": "2021",
      "poststed": "JAJAJA"
    },
    "datakilder": [{}, {}]
  }
}
