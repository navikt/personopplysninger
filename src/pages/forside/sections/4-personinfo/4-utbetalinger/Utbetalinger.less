.utbetalinger {
  &__rad {
    display: flex;
    @media @maxMobile {
      flex-direction: column;
    }
  }
  &__type {
    margin-bottom: 1rem;

    .inputPanelGruppe__inner {
      @media @minMobile {
        display: flex;
        > * {
          width: 50%;
          &:first-child {
            margin-right: 0.5rem;
          }
        }
      }
    }
  }
  &__bank-identifier {
    margin: 1rem 0;
  }
  &__knapper {
    display: flex;
    padding: 1rem 0;
  }
  &__knapp:last-child {
    margin-left: 0.5rem;
  }
  &__input-container {
    display: flex;
    flex-wrap: wrap;
  }
  &__alert {
    padding: 0.5rem 0 1rem 0;
  }
  &__input-box {
    @media @minMobile {
      min-width: 50%;
      &:nth-child(odd) {
        padding-right: 0.25rem;
      }
      &:nth-child(even) {
        padding-left: 0.25rem;
      }
    }
    @media @maxMobile {
      min-width: 100%;
    }
  }

  &__adressefelter {
    width: 100%;
  }
  &__bankkode {
    &-rad {
      display: flex;
      margin-bottom: 1.5rem;
    }
    &-kolonne:first-child {
      max-width: 4rem;
      margin-right: 0.15rem;
    }
    &-kolonne:last-child {
      width: 100%;

      .navds-form-field {
        gap: 0;
      }
    }
  }
}
