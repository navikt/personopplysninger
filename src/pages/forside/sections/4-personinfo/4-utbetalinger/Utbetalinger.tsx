import React from "react";
import { UtenlandskBankkonto } from "types/personalia";
import { useStore } from "providers/Provider";
import UtbetalingerPDL from "./Utbetalinger-PDL";
import UtbetalingerOLD from "./Utbetalinger-OLD";

interface Props {
  utenlandskbank?: UtenlandskBankkonto;
  kontonr?: string;
}
const Utbetalinger = (props: Props) => {
  const [{ featureToggles }] = useStore();
  return featureToggles.data["personopplysninger.pdl"] ? (
    <UtbetalingerPDL
      kontonr={props.kontonr}
      utenlandskbank={props.utenlandskbank}
    />
  ) : (
    <UtbetalingerOLD
      kontonr={props.kontonr}
      utenlandskbank={props.utenlandskbank}
    />
  );
};

export default Utbetalinger;
