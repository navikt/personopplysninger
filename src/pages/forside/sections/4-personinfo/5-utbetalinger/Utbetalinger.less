.utbetalinger {
  &__type {
    margin-bottom: 1rem;

    .inputPanelGruppe__inner {
      display: flex;
      > * {
        width: 50%;
        &:first-child {
          margin-right: 0.5rem;
        }
      }
    }
  }
  &__knapp-container {
    display: flex;
    justify-content: center;
  }
}
