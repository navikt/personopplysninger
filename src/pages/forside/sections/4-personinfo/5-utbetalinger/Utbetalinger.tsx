import React, { Component } from "react";
import Box from "../../../../../components/box/Box";
import kontonummerIkon from "../../../../../assets/img/Kontonummer.svg";
import { Personalia as PersonaliaType } from "../../../../../types/personalia";
import Kilde from "../../../../../components/kilde/Kilde";
import endreIkon from "../../../../../assets/img/Pencil.svg";
import Environment from "../../../../../utils/Environments";
import Melding from "../../../../../components/melding/Melding";
import NorskKontonummer from "./visning/NorskKontonummer";
import Utenlandskonto from "./visning/UtenlandsBankkonto";

interface Props {
  personalia: PersonaliaType;
}

const { tjenesteUrl } = Environment();

class Utbetalinger extends Component<Props> {
  render() {
    const { kontonr, utenlandskbank } = this.props.personalia;

    return (
      <Box id="utbetaling" tittel="utbetalinger.tittel" icon={kontonummerIkon}>
        <hr className="box__linje-bred" />
        {kontonr || utenlandskbank ? (
          <>
            <NorskKontonummer kontonummer={kontonr} />
            <Utenlandskonto utenlandskBankkonto={utenlandskbank} />
          </>
        ) : (
          <Melding meldingId="personalia.kontonr.ingenData" />
        )}
        <Kilde
          kilde="personalia.source.nav"
          lenke={`${tjenesteUrl}/brukerprofil/`}
          lenkeTekst="personalia.link.brukerprofil.endre"
          lenkeType={"EKSTERN"}
          ikon={endreIkon}
        />
      </Box>
    );
  }
}

export default Utbetalinger;
