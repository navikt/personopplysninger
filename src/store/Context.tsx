import { createContext, useContext, useReducer, Dispatch } from 'react';
import { Action, initialState, reducer, Store } from './Store';
interface Props {
    children: JSX.Element | JSX.Element[];
}

export const StoreContext = createContext({} as [Store, Dispatch<Action>]);
export const StoreProvider = (props: Props) => {
    const { children } = props;

    return <StoreContext.Provider value={useReducer(reducer, initialState)}>{children}</StoreContext.Provider>;
};
export const useStore = () => useContext(StoreContext);
