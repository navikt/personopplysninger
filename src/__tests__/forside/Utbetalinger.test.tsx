import '@testing-library/jest-dom';
import { IntlProvider } from 'react-intl';
import nbMessages from 'text/nb';
import personInfo from '../../clients/apiMock/app/fetch/person-info.json';
import personInfoUtenlandskbank from '../../clients/apiMock/app/fetch/person-info-utenlandskbank.json';
import Utbetalinger from '../../pages/forside/sections/4-personinfo/4-utbetalinger/Utbetalinger';
import { StoreProvider } from '../../store/Context';
import { render } from '@testing-library/react';
import { UtenlandskBankkonto } from '../../types/personalia';
import { MemoryRouter } from 'react-router-dom';

jest.mock('react-modal');

describe('Utbetalinger', () => {
    it('with kontonummer renders correctly', () => {
        const { asFragment } = render(
            <StoreProvider>
                <IntlProvider locale={'nb'} messages={nbMessages}>
                    <MemoryRouter>
                        <Utbetalinger kontonr={personInfo.personalia.kontonr} kontoregisterStatus={'OK'} />
                    </MemoryRouter>
                </IntlProvider>
            </StoreProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('with utenlandsk bank renders correctly', () => {
        const { asFragment } = render(
            <StoreProvider>
                <IntlProvider locale={'nb'} messages={nbMessages}>
                    <MemoryRouter>
                        <Utbetalinger
                            utenlandskbank={personInfoUtenlandskbank.personalia.utenlandskbank as unknown as UtenlandskBankkonto}
                            kontoregisterStatus={'OK'}
                        />
                    </MemoryRouter>
                </IntlProvider>
            </StoreProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
