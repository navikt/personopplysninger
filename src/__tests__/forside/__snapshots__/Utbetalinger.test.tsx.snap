// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utbetalinger with kontonummer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="box__wrapper"
    id="utbetaling"
  >
    <div
      class="navds-guide-panel navds-guide-panel--poster"
    >
      <div
        class="navds-guide"
      >
        <img
          alt=""
          class="box__ikon"
          src="Kontonummer.svg"
        />
      </div>
      <div
        class="navds-guide-panel__content"
      >
        <div
          class="box__container"
        >
          <div
            class="box__header"
          >
            <div
              class="box__title-container"
            >
              <div
                class="box__line"
              />
              <h2
                class="box__title navds-heading navds-heading--medium"
              >
                Kontonummer for utbetaling
              </h2>
              <div
                class="box__line"
              />
            </div>
            <a
              class="navds-link anchor-link navds-link--action"
              href="#utbetaling"
            >
              <svg
                aria-hidden="true"
                class="anchor-link__icon"
                fill="none"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.909 5.152a2.25 2.25 0 0 1 3.182 0l.757.757a2.25 2.25 0 0 1 0 3.182l-4.964 4.965a1.25 1.25 0 0 1-1.768 0L11.03 12.97a.75.75 0 1 0-1.06 1.06l1.085 1.086a2.75 2.75 0 0 0 3.89 0l4.964-4.964a3.75 3.75 0 0 0 0-5.304l-.757-.757-.53.53.53-.53a3.75 3.75 0 0 0-5.304 0l-.878.879a.75.75 0 0 0 1.06 1.06l.879-.878Zm-1.964 3.732a2.75 2.75 0 0 0-3.89 0l-4.964 4.964a3.75 3.75 0 0 0 0 5.304l.513-.514-.513.514.757.757a3.75 3.75 0 0 0 5.304 0l.878-.879a.75.75 0 0 0-1.06-1.06l-.879.878a2.25 2.25 0 0 1-3.182 0l-.757-.757a2.25 2.25 0 0 1 0-3.182l4.964-4.965a1.25 1.25 0 0 1 1.768 0l1.086 1.086a.75.75 0 0 0 1.06-1.06l-1.085-1.086Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              Lenke hit
            </a>
          </div>
          <div
            class="box__content"
          >
            <dl
              class="list"
            >
              <div
                class="list-item"
              >
                <dt
                  class="list-element-heading"
                >
                  <span
                    class="navds-label"
                  >
                    Kontonummer
                  </span>
                </dt>
                <dd>
                  1234 56 78911
                </dd>
              </div>
            </dl>
            <div
              class="kilde__container"
            >
              <div
                class="kilde__tekst"
              >
                <p
                  class="navds-detail"
                >
                  Kilde: NAV
                </p>
              </div>
              <button
                class="kilde__knapp knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
              >
                <span
                  class="navds-label"
                >
                  <span
                    class="kilde__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="kilde__icon-aksel"
                      fill="none"
                      focusable="false"
                      height="1em"
                      role="img"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M19.638 4.417a3.252 3.252 0 0 0-4.608-.008l-9.378 9.379a.75.75 0 0 0-.191.324l-1.414 4.95a.75.75 0 0 0 .925.927l4.94-1.398a.75.75 0 0 0 .327-.191l9.39-9.391a3.252 3.252 0 0 0 .01-4.592ZM16.091 5.47a1.752 1.752 0 0 1 2.478 2.478l-.23.23-2.477-2.479.229-.23ZM14.8 6.76 6.85 14.71l-.991 3.47 3.457-.979 7.963-7.963-2.478-2.478Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  Endre
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Utbetalinger with utenlandsk bank renders correctly 1`] = `
<DocumentFragment>
  <div
    class="box__wrapper"
    id="utbetaling"
  >
    <div
      class="navds-guide-panel navds-guide-panel--poster"
    >
      <div
        class="navds-guide"
      >
        <img
          alt=""
          class="box__ikon"
          src="Kontonummer.svg"
        />
      </div>
      <div
        class="navds-guide-panel__content"
      >
        <div
          class="box__container"
        >
          <div
            class="box__header"
          >
            <div
              class="box__title-container"
            >
              <div
                class="box__line"
              />
              <h2
                class="box__title navds-heading navds-heading--medium"
              >
                Kontonummer for utbetaling
              </h2>
              <div
                class="box__line"
              />
            </div>
            <a
              class="navds-link anchor-link navds-link--action"
              href="#utbetaling"
            >
              <svg
                aria-hidden="true"
                class="anchor-link__icon"
                fill="none"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.909 5.152a2.25 2.25 0 0 1 3.182 0l.757.757a2.25 2.25 0 0 1 0 3.182l-4.964 4.965a1.25 1.25 0 0 1-1.768 0L11.03 12.97a.75.75 0 1 0-1.06 1.06l1.085 1.086a2.75 2.75 0 0 0 3.89 0l4.964-4.964a3.75 3.75 0 0 0 0-5.304l-.757-.757-.53.53.53-.53a3.75 3.75 0 0 0-5.304 0l-.878.879a.75.75 0 0 0 1.06 1.06l.879-.878Zm-1.964 3.732a2.75 2.75 0 0 0-3.89 0l-4.964 4.964a3.75 3.75 0 0 0 0 5.304l.513-.514-.513.514.757.757a3.75 3.75 0 0 0 5.304 0l.878-.879a.75.75 0 0 0-1.06-1.06l-.879.878a2.25 2.25 0 0 1-3.182 0l-.757-.757a2.25 2.25 0 0 1 0-3.182l4.964-4.965a1.25 1.25 0 0 1 1.768 0l1.086 1.086a.75.75 0 0 0 1.06-1.06l-1.085-1.086Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              Lenke hit
            </a>
          </div>
          <div
            class="box__content"
          >
            <dl
              class="list"
            >
              <div
                class="list-item"
              >
                <dt
                  class="list-element-heading"
                >
                  <span
                    class="navds-label"
                  >
                    Bank
                  </span>
                </dt>
                <dd>
                  <p
                    class="navds-body-short navds-body-short--medium"
                  >
                    Dominion Bank
                  </p>
                  <p
                    class="navds-body-short navds-body-short--medium"
                  >
                    Forente Arabiske Emirater
                  </p>
                </dd>
              </div>
              <div
                class="list-item"
              >
                <dt
                  class="list-element-heading"
                >
                  <span
                    class="navds-label"
                  >
                    Kontonummer / IBAN
                  </span>
                </dt>
                <dd>
                  AE460090000000123456789
                </dd>
              </div>
              <div
                class="list-item"
              >
                <dt
                  class="list-element-heading"
                >
                  <span
                    class="navds-label"
                  >
                    Valuta
                  </span>
                </dt>
                <dd>
                  Amerikansk dollar
                </dd>
              </div>
              <div
                class="list-item"
              >
                <dt
                  class="list-element-heading"
                >
                  <span
                    class="navds-label"
                  >
                    BIC / SWIFT-kode
                  </span>
                </dt>
                <dd>
                  EBILAEADXXX
                </dd>
              </div>
            </dl>
            <div
              class="kilde__container"
            >
              <div
                class="kilde__tekst"
              >
                <p
                  class="navds-detail"
                >
                  Kilde: NAV
                </p>
              </div>
              <button
                class="kilde__knapp knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
              >
                <span
                  class="navds-label"
                >
                  <span
                    class="kilde__icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="kilde__icon-aksel"
                      fill="none"
                      focusable="false"
                      height="1em"
                      role="img"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M19.638 4.417a3.252 3.252 0 0 0-4.608-.008l-9.378 9.379a.75.75 0 0 0-.191.324l-1.414 4.95a.75.75 0 0 0 .925.927l4.94-1.398a.75.75 0 0 0 .327-.191l9.39-9.391a3.252 3.252 0 0 0 .01-4.592ZM16.091 5.47a1.752 1.752 0 0 1 2.478 2.478l-.23.23-2.477-2.479.229-.23ZM14.8 6.76 6.85 14.71l-.991 3.47 3.457-.979 7.963-7.963-2.478-2.478Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  Endre
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
